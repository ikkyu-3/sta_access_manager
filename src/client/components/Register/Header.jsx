// @flow
import type { RegisterHeaderClassName, RegisterHeaderProps } from "@types";

import * as React from "react";
import { INPUT, SCAN, COMPLETION } from "@constants";
import styles from "./style.css";

function getClassName(registerStep: string): RegisterHeaderClassName {
  const { active, disabled } = styles;

  if (registerStep === INPUT) {
    return { input: active, scan: disabled, completion: disabled };
  }

  if (registerStep === SCAN) {
    return { scan: active, completion: disabled };
  }

  if (registerStep === COMPLETION) {
    return { completion: active };
  }

  return { input: disabled, scan: disabled, completion: disabled };
}

function Header(props: RegisterHeaderProps): React.Node {
  const { registerStep } = props;
  const { header } = styles;
  const { input, scan, completion } = getClassName(registerStep);
  return (
    <ol className={header}>
      <li className={input}>
        1. 入力<span>&gt;</span>
      </li>
      <li className={scan}>
        2. 読取<span>&gt;</span>
      </li>
      <li className={completion}>3. 完了</li>
    </ol>
  );
}
export default Header;
