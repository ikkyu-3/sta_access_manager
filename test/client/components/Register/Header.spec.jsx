import React from "react";
import renderer from "react-test-renderer";
import { shallow } from "enzyme";
import Header from "@components/Register/Header";

describe("Register/Header.jsxのテスト", () => {
  describe("スナップショット", () => {
    it("正しいレンダリング", () => {
      const tree = renderer.create(<Header registerStep="input" />);
      expect(tree).toMatchSnapshot();
    });
  });

  describe("コンポーネントのテスト", () => {
    describe("registerStepにinputを指定した時", () => {
      let header;

      beforeAll(() => {
        header = shallow(<Header registerStep="input" />).find("li");
      });

      it("li要素の0番目の子要素は、cssクラス: activeを持つ", () => {
        expect(header.at(0).hasClass("active")).toBeTruthy();
      });

      it("li要素の1番目の子要素は、cssクラス: disabledを持つ", () => {
        expect(header.at(1).hasClass("disabled")).toBeTruthy();
      });

      it("li要素の2番目の子要素は、cssクラス: disabledを持つ", () => {
        expect(header.at(2).hasClass("disabled")).toBeTruthy();
      });
    });

    describe("registerStepにscanを指定した時", () => {
      let header;

      beforeAll(() => {
        header = shallow(<Header registerStep="scan" />).find("li");
      });

      it("li要素の0番目の子要素は、クラスを持たない(classNameは、undifined)", () => {
        expect(header.at(0).props().className).toBeUndefined();
      });

      it("li要素の1番目の子要素は、cssクラス: activeを持つ", () => {
        expect(header.at(1).hasClass("active")).toBeTruthy();
      });

      it("li要素の2番目の子要素は、cssクラス: disabledを持つ", () => {
        expect(header.at(2).hasClass("disabled")).toBeTruthy();
      });
    });

    describe("registerStepにcompletionを指定した時", () => {
      let header;

      beforeAll(() => {
        header = shallow(<Header registerStep="completion" />).find("li");
      });

      it("li要素の0番目の子要素は、クラスを持たない(classNameは、undifined)", () => {
        expect(header.at(0).props().className).toBeUndefined();
      });

      it("li要素の1番目の子要素は、クラスを持たない(classNameは、undifined)", () => {
        expect(header.at(1).props().className).toBeUndefined();
      });

      it("li要素の2番目の子要素は、cssクラス: activeを持つ", () => {
        expect(header.at(2).hasClass("active")).toBeTruthy();
      });
    });

    describe("currentに値を指定しなかった時", () => {
      let header;

      beforeAll(() => {
        header = shallow(<Header registerStep="" />).find("li");
      });

      it("li要素の0番目の子要素は、クラスを持たない(classNameは、disabled)", () => {
        expect(header.at(0).hasClass("disabled")).toBeTruthy();
      });

      it("li要素の1番目の子要素は、クラスを持たない(classNameは、undifined)", () => {
        expect(header.at(1).hasClass("disabled")).toBeTruthy();
      });

      it("li要素の2番目の子要素は、クラスを持たない(classNameは、undifined)", () => {
        expect(header.at(2).hasClass("disabled")).toBeTruthy();
      });
    });
  });
});
