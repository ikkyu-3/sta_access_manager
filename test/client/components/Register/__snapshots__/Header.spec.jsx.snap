// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Register/Header.jsxのテスト スナップショット 正しいレンダリング 1`] = `
<ol
  className="header"
>
  <li
    className="active"
  >
    1. 入力
    <span>
      &gt;
    </span>
  </li>
  <li
    className="disabled"
  >
    2. 読取
    <span>
      &gt;
    </span>
  </li>
  <li
    className="disabled"
  >
    3. 完了
  </li>
</ol>
`;
